src/routing/mod.rs

```diff
@@ -400,14 +400,47 @@ impl Routing {
     async fn send_message_to_client(
         &self,
         recipient: SocketAddr,
-        message: ClientMessage,
+        message: ClientMsg,
     ) -> Result<()> {
-        let command = Command::SendMessage {
-            recipients: vec![recipient],
-            delivery_group_size: 1,
-            message: MessageType::ClientMessage(message),
+        let end_user = self
+            .dispatcher
+            .core
+            .lock()
+            .await
+            .get_enduser_by_addr(&recipient)
+            .copied();
+        let end_user_pk = match end_user {
```

Comment:
This function is being called by `send_message` , which shall already have the `public_key` from the `itinerary.dst` at the calling place.
why not just pass that in, instead carry out another round of fetch?

Reply:
We could do even better here, since we are the client's own section(proved by us having the client's socket_addr in calling function), we can just simply add our section PK instead of matching and finding. Will update 👍